From 89acf26de991c67dc4ab5df353f50afc30f07748 Mon Sep 17 00:00:00 2001
From: David Kahurani <redress@duck.com>
Date: Sat, 10 May 2025 14:17:39 +0300
Subject: [PATCH] io_uring: disallow the use of cloning to register arbitrarily
 large buffers

Currently this allows for users to register buffers larger
than the allowed maximum

Signed-off-by: David Kahurani <redress@duck.com>
---
 io_uring/rsrc.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/io_uring/rsrc.c b/io_uring/rsrc.c
index dbb94d932..e80fba1f5 100644
--- a/io_uring/rsrc.c
+++ b/io_uring/rsrc.c
@@ -975,6 +975,10 @@ static int io_clone_buffers(struct io_ring_ctx *ctx, struct io_ring_ctx *src_ctx
 		return -EINVAL;
 	else if (arg->nr > IORING_MAX_REG_BUFFERS)
 		return -EINVAL;
+
+	if (arg->dst_off > IORING_MAX_REG_BUFFERS)
+		return EINVAL;
+
 	if (check_add_overflow(arg->nr, arg->dst_off, &nbufs))
 		return -EOVERFLOW;
 
-- 
2.43.0

